libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -235,7 +235,12 @@ proc handleSubscribe*(g: GossipSub,
 
     # unsubscribe remote peer from the topic
     g.gossipsub.removePeer(topic, peer)
-    g.mesh.removePeer(topic, peer)
+
+    if g.mesh.hasPeer(topic, peer):
```

Comment:
since the proper order of events  is "remove from mesh" then "remove from gossipsub" I'd keep it that way in this code as well.. this gives "balance" when reading the code in general (sub -> graft -> prune -> unsub)

Reply:
Done